rosbag2_to_video/api.py

```diff
@@ -30,25 +30,45 @@ def get_stamp_from_image_msg(image_msg):
 
 def add_arguments_to_parser(argparser):
     argparser.add_argument(
-        'bagfile', help='Path to the bag (currently only supports sensor_msgs/msg/Image types)')
-    argparser.add_argument('-t','--topic', required=True, help='Name of the image topic')
-    argparser.add_argument('-o','--output', required=True, help='Output filename (currently only supports MP4)')
+        'bagfile',
+        help='Path to the bag (currently only supports sensor_msgs/msg/Image or sensor_msgs/msg/CompressedImage types)' # noqa E501
+    )
+    argparser.add_argument(
+        '-t',
+        '--topic',
+        required=True,
+        help='Name of the image topic'
+    )
+    argparser.add_argument(
+        '-o',
+        '--output',
+        required=True,
+        help='Output filename (currently only supports MP4)'
+    )
     argparser.add_argument('--fps', type=int, required=True, help='Output frames per second')
+    argparser.add_argument('--storage-id', type=str, default="sqlite3", help='Rosbag2 storage id')
 
 
 def convert_bag_to_video(args):
     # Force the .mp4 extension on file output name
     if args.output[-4:] != '.mp4':
         args.output += '.mp4'
 
-    storage_options = rosbag2_py.StorageOptions(uri=args.bagfile)
+    storage_options = rosbag2_py.StorageOptions(uri=args.bagfile, storage_id=args.storage_id)
     # TODO(jacobperron): Shouldn't we be able to infer serialization format from metadaya.yaml?
     converter_options = rosbag2_py.ConverterOptions(
         input_serialization_format='cdr', output_serialization_format='cdr')
 
     bag_reader = rosbag2_py.SequentialReader()
     bag_reader.open(storage_options, converter_options)
 
+    topic_types = bag_reader.get_all_topics_and_types()
+    type_map = {topic_types[i].name: topic_types[i].type for i in range(len(topic_types))}
+    print(type_map[args.topic])
+    if (type_map[args.topic] != 'sensor_msgs/msg/Image' and
+            type_map[args.topic] != 'sensor_msgs/msg/CompressedImage'):
+        raise ValueError('topic type should be sensor_msgs/msg/Image or sensor_msgs/msg/CompressedImage') # noqa E501
```

Comment:
You need the topic type for `args.topic`, not for anything else.
So constructing a map seems unnecessary.

```suggestion
    try:
        topic_type = next(x for x in topic_types if x.name == args.topic).type
    except StopIteration:
        print(f'Topic {args.topic} was not recorded in the bagfile')
        return
    if topic_type not in ('sensor_msgs/msg/Image', 'sensor_msgs/msg/CompressedImage'):
          print(
              'topic type should be sensor_msgs/msg/Image or '
              f'sensor_msgs/msg/CompressedImage, got {topic_type}')
```